src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseBase.cpp

```diff
@@ -262,9 +279,114 @@ void SinglePhaseBase::initializePostInitialConditionsPreSubGroups()
       arrayView1d< real64 > const presInit   = subRegion.getExtrinsicData< extrinsicMeshData::flow::initialPressure >();
       presInit.setValues< parallelDevicePolicy<> >( pres );
     } );
+
+    // If requested by the user, compute some statistics per region
+    if( m_computeStatistics )
+    {
+      computeRegionStatistics( mesh, regionNames );
+    }
+
+  } );
+}
+
+void SinglePhaseBase::computeRegionStatistics( MeshLevel & mesh,
+                                               arrayView1d< string const > const & regionNames ) const
+{
+  GEOSX_MARK_FUNCTION;
+
+  // Step 1: initialize the average/min/max quantities
+  ElementRegionManager & elemManager = mesh.getElemManager();
+  for( integer i = 0; i < regionNames.size(); ++i )
+  {
+    ElementRegionBase & region = elemManager.getRegion( regionNames[i] );
+
+    real64 & avgPres = region.getReference< real64 >( viewKeyStruct::averagePressureString() );
+    real64 & minPres = region.getReference< real64 >( viewKeyStruct::minimumPressureString() );
+    real64 & maxPres = region.getReference< real64 >( viewKeyStruct::maximumPressureString() );
+    avgPres = 0.0; maxPres = 0.0; minPres = LvArray::NumericLimits< real64 >::max;
+
+    real64 & totalPoreVol = region.getReference< real64 >( viewKeyStruct::totalPoreVolumeString() );
+    real64 & totalUncompactedPoreVol = region.getReference< real64 >( viewKeyStruct::totalUncompactedPoreVolumeString() );
+    totalPoreVol = 0.0; totalUncompactedPoreVol = 0.0;
+  }
+
+  // Step 2: increment the average/min/max quantities for all the subRegions
+  elemManager.forElementSubRegions( regionNames, [&]( localIndex const,
+                                                      ElementSubRegionBase & subRegion )
+  {
+
+    arrayView1d< integer const > const elemGhostRank = subRegion.ghostRank();
+    arrayView1d< real64 const > const volume = subRegion.getElementVolume();
+    arrayView1d< real64 const > const pres = subRegion.getExtrinsicData< extrinsicMeshData::flow::pressure >();
+
+    string const & solidName = subRegion.getReference< string >( viewKeyStruct::solidNamesString() );
+    CoupledSolidBase const & solid = getConstitutiveModel< CoupledSolidBase >( subRegion, solidName );
+    arrayView1d< real64 const > const refPorosity = solid.getReferencePorosity();
+    arrayView2d< real64 const > const porosity = solid.getPorosity();
+
+    real64 subRegionAvgPresNumerator = 0.0;
+    real64 subRegionMinPres = 0.0;
+    real64 subRegionMaxPres = 0.0;
+    real64 subRegionTotalUncompactedPoreVol = 0.0;
+    real64 subRegionTotalPoreVol = 0.0;
```

Comment:
Do these values really make sense?

Reply:
Yes, it makes sense to initialize the values like this before computing the min and the max over the sub-region